### ISSUE TITLE
Read PULUMI_STACK, etc from env variables

### ISSUE THREAD

**Issue by bigluck** [pulumi/pulumi#13550]
I would like to control the pulumi execution context using environment variables.

There are several use cases where it is more convenient to have a `PULUMI_STACK=my/stack pulumi preview` instead of `pulumi preview -s my/stack`.

In an advanced use case the pulumi execution context in a CI/CD can be controlled using environment variables without having to explicitly pass the value as an argument.

At the same time, this approach can be extended to a local development environment by using direnv and a custom hook that both activate the AWS session (using assume or any other 3rd party tools) and exports (by reading a .env file) the appropriate pulumi stack to use.

Looks like golang has some "kind" of support for some envs, but I wasn't able to trigger it using my cli

```bash
$ pulumi stack
Please choose a stack, or create a new one:  [Use arrows to move, type to filter]
  my/stack_1
  my/stack_2
> <create a new stack>
```

```bash
$ PULUMI_STACK=my/stack_1 pulumi preview
Please choose a stack, or create a new one:  [Use arrows to move, type to filter]
  my/stack_1
  my/stack_2
> <create a new stack>
```

```bash
$ pulumi preview -s my/stack_1
# it works!
```

Any suggest?

Thanks so much.
Luca

**Comment by Frassle** [comment]
Currently the only "global" place the current stack is pulled from is in ~/.pulumi/workspaces, but that's not easy to write to (the folder names are all hashed).

It doesn't seem unreasonable to allow some config like this to be environment rather than just by settings files.

**Comment by PierreBeucher** [comment]
Having a `PULUMI_STACK` environment would be very useful indeed, and would go nicely with  `PULUMI_CONFIG_PASSPHRASE` to have tuple stack / passphrase

**Comment by olee** [comment]
I fully agree. I was really surprised to find out that PULUMI_STACK is not a thing yet as it would make building things like github workflows a lot easier.

**Comment by aloeenmae** [comment]
Still no updates?

**Comment by sayadi** [comment]
This is going to live in my terminal profile until we get this officially!

```
alias pulumi='pulumi -s $PULUMI_STACK'
```
